### DISCUSSION TITLE
Use of weak ciphers

### DISCUSSION THREAD

**Discussion by alexxsanya** [invertase/react-native-firebase#5176] [General]
Hello,

First of all, thanks for the good work on this package. We use it in our enterprise app - we love it.

Recently, we scanned for security issues in the app and this came up. This package was found to be using weak ciphers (MD5 and/or SHA1). See below.

![image](https://user-images.githubusercontent.com/21230051/114787070-d420d780-9d87-11eb-8d51-5e63d0b7b495.png)

The report recommends that one should use stronger ciphers like SHA256. Is this something you plan on looking into in the near future?

**Comment by mikehardy** [comment]
We simply wrap the underlying firebase-ios-sdk and firebase-android-sdk libraries, while matching the firebase-js-sdk API as much as possible as we do so.

Those libraries use an md5 hash for firebase storage files.

https://firebase.google.com/docs/reference/js/firebase.storage.FullMetadata#md5hash

You'll want to take this up with them, as we will have it as long as they do

**Reply by alexxsanya** [reply]
Thank you so much @mikehardy for clarifying this.
